server/src/routes.ts

```diff
@@ -2215,6 +2266,480 @@ export default async function routes(fastify: FastifyInstance) {
+                header_color: { type: 'string' },
+                header_link_color: { type: 'string' },
+                main_color: { type: 'string' },
+                footer_color: { type: 'string' },
+                left_image_url: { type: 'string' },
+                left_image_link: { type: 'string' },
+                right_image_url: { type: 'string' },
+                right_image_link: { type: 'string' },
+                mobile_image_url: { type: 'string' },
+                mobile_image_link: { type: 'string' },
+                footer_icon: { type: 'string' },
+                secret_code: { type: 'string' },
+                created_date: { type: 'string' },
+                is_active: { type: 'boolean' },
+              },
+            }
+          },
+          security: [
+            {
+              "authorization": []
+            }
+          ]
+        },
+      },
+      async (req, res) => {
+        const event_template = await getEventTemplate(req.params.id);
+        if (!event_template) {
+          return new createError.NotFound('Invalid Event Template');
+        }
+        return event_template;
+      }
+  );
+
+  async function validate_template_images(req: any) {
+    const title_image = { ...req.body.title_image };
+    req.body.title_image = '@title_image';
+    req.body[Symbol.for('title_image')] = title_image;
+
+    const right_image_url = { ...req.body.right_image_url };
+    req.body.right_image_url = '@right_image_url';
+    req.body[Symbol.for('right_image_url')] = right_image_url;
+
+    const left_image_url = { ...req.body.left_image_url };
+    req.body.left_image_url = '@left_image_url';
+    req.body[Symbol.for('left_image_url')] = left_image_url;
+
+    const mobile_image_url = { ...req.body.mobile_image_url };
+    req.body.mobile_image_url = '@mobile_image_url';
+    req.body[Symbol.for('mobile_image_url')] = mobile_image_url;
+
+    const footer_icon = { ...req.body.footer_icon };
+    req.body.footer_icon = '@footer_icon';
+    req.body[Symbol.for('footer_icon')] = footer_icon;
+  }
+
+  fastify.post(
+      '/event-templates',
+      {
+        preValidation: [fastify.optionalAuthenticate, validate_template_images,],
+        schema: {
+          description: 'Endpoint to create new event templates',
+          tags: ['Event Templates',],
+          body: {
+            type: 'object',
+            required: [
+              'name',
+              'title_image',
+              'title_link',
+              'header_link_text',
+              'header_link_url',
+              'header_color',
+              'header_link_color',
+              'main_color',
+              'footer_color',
+              'left_image_url',
+              'left_image_link',
+              'right_image_url',
+              'right_image_link',
+              'mobile_image_url',
+              'mobile_image_link',
+              'footer_icon',
+              'secret_code'
```

Comment:
We missed talking about required fields. They are the following:
```
'name',
'title_image',
'title_link',
'header_color',
'header_link_color',
'main_color',
'footer_icon',
'secret_code'
```

Please review validations and refactor